pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -359,18 +364,23 @@ func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, branch
 	}
 
 	// look for a ProcessActivityNode by process cookie
-	if entry.Cookie > 0 {
+	cs := cookieSelector{}
+	if entry.ExecTime.UnixNano() != 0 && entry.Cookie != 0 {
+		cs = cookieSelector{
+			execTime: entry.ExecTime.UnixNano(),
+			cookie:   entry.Cookie,
+		}
 		var found bool
-		node, found = at.CookieToProcessNode[entry.Cookie]
+		node, found = at.CookieToProcessNode[cs]
 		if found {
 			return node, false, nil
 		}
 	}
 
 	defer func() {
 		// if a node was found, and if the entry has a valid cookie, insert a cookie shortcut
-		if entry.Cookie > 0 && node != nil {
-			at.CookieToProcessNode[entry.Cookie] = node
+		if cs.cookie != 0 && node != nil {
```

Comment:
I think you need to check both the exec time and the cookie again here :( => should we create a method for `cookieSelector` ?

Reply:
Suggestion applied, thanks